composer/utils/batch_helpers.py

```diff
@@ -211,12 +189,11 @@ def _batch_set_tuple(batch: Any, key: Union[int, str], value: Any) -> Any:
     return batch
 
 
-def _batch_get_tuple(batch: Any, key: Union[int, str]) -> Any:
-    """"Gets keys in tuples and NamedTuples."""
-    is_named_tuple = hasattr(batch, '_fields')
-    if is_named_tuple and isinstance(key, str):  # NamedTuple w/ named key
-        value = getattr(batch, key)
-    else:  # Normal tuple or namedtuple with int key.
-        value = batch[key]
-
-    return value
+def _is_key_get_and_set_fn_pair(key):
+    if isinstance(key, Sequence) and not isinstance(key, str):
+        if all([isinstance(key_element, Callable) for key_element in key]):
```

Comment:
Nit: prefer to use the `callable()` check instead of `isinstance` -- while it may work here, usually `isinstance` checks on imports from `typing` do not work.

```suggestion
        if all([callable(key_element) for key_element in key]):
```